### PR TITLE
Add base code and implement "add" and "view" feature for todo list

### LINKED ISSUE #157: TodoList : Add base code and implement "add task" and "view task" feature

### PR DESCRIPTION
**Pull request by michael-setia**
Fixes #157 
- Initial implementation for todo list.
- Added todo list to the main logic.
- Added 2 more prefixes :
  - "t/" for input that represents a type.
  - "p/" for input that represents a priority.
- 2 Prefixes are reused :
  - "n/" is reused to represent description/name of a task
  - "d/" is reused to represent date/deadline of a task
- Sample command:
  - addtask n/finish V.2 t/project d/2020-10-07 p/high
  - viewtodolist

### CODE REVIEW

File: src/main/java/tp/cap5buddy/todolist/ParserUtilTodoList.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        switch(inputAllUpperCase) {
+        case("DESCRIPTION"):
+        case("DESC"):
+            return Criterion.DESCRIPTION;
+        case("DATE"):
+        case("DEADLINE"):
+            return Criterion.DATE;
+        case("PRIORITY"):
+        case("PRIO"):
+            return Criterion.PRIORITY;
```
Comment (murtubak) [inline src/main/java/tp/cap5buddy/todolist/ParserUtilTodoList.java:121]: Is this just for short form?

**Review by murtubak — COMMENTED** [review]
LGTM, but should the tasks and todolist be stored with the modules as JsonAdaptedTasks

File: src/main/java/tp/cap5buddy/todolist/ParserUtilTodoList.java (hunk 1 above)
Comment (michael-setia) [reply src/main/java/tp/cap5buddy/todolist/ParserUtilTodoList.java:121]: Yes, for now it is just used to make the testing easier (no need to write the whole thing every time).

**Review by murtubak — APPROVED** [review]
LGTM.

**Comment by michael-setia** [comment]
Yes, I will implement the storage part for TodoList later.

**Pull request merged**